src/plugins/auto/utils/log.hpp

```diff
@@ -241,8 +241,15 @@ inline void Log::doLog(bool on, bool isTraceCallStack, LogLevel level, const cha
         stream << '[' << tag << ']';
     }
     char buffer[255];
-    checkFormat(fmt);
-    std::string compatibleString =  "%s" + std::string(fmt);
+    std::string compatibleString;
+
+    try {
+        checkFormat(fmt);
+        compatibleString =  "%s" + std::string(fmt);
+    } catch (std::runtime_error &err) {
+        compatibleString =  "%s" + std::string(err.what());
+    }
+
     std::snprintf(&buffer[0], sizeof(buffer), compatibleString.c_str(), "", args...);
```

Comment:
if the format is wrong, I suggest checkFormat return err message if thrown runtime error is not permitted,  and snprintf will print this err message instead of origin log